mingxguo/assignment-3/lexicon.h

```diff
@@ -0,0 +1,26 @@
+#ifndef MINGXGUO_ASSIGNMENT3_LEXICON_H_
+#define MINGXGUO_ASSIGNMENT3_LEXICON_H_
+
+#include <iostream>
+#include <set>
+#include <string>
+
+using namespace std;
+
+class Lexicon{
+public:
+	// Data structure that represent the lexicon (an ordered set of words)
+	set<string> lexicon;
+
+public:
+	// Empty constructor
+	Lexicon();
+	// Add word to the lexicon
+	void AddWord(string const& word);

```

Comment:
To be able to pass the lexicon as const where appropriate, you should mark the methods that don't modify the class variables as const too. Marking the functions as const means your code won't compile if you change the class variables somehow in them. That way you make sure the lexicon really doesn't change when calling some method.

```suggestion
	void AddWord(string const& word) const;
```